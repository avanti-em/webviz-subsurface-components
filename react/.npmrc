engine-strict=true
ignore-scripts=true
CYPRESS_INSTALL_BINARY=0