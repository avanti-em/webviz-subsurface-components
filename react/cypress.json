{
    "baseUrl": "http://localhost:6006/?path=/story/deckglmap--default",
    "video": true,
    "screenshotsFolder": "./cypress/snapshots/actual",
    "trashAssetsBeforeRuns": true,
    "env": {
        "failSilently": false,
        "ALWAYS_GENERATE_DIFF": false,
        "type": "base"
    }
}
