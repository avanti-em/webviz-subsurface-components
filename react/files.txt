cypress-visual-screenshots/comparison/defaultMap.cy-default-map-story.png
cypress-visual-screenshots/diff/defaultMap.cy-default-map-story.png
files.txt
