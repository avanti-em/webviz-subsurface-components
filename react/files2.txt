cypress-visual-screenshots/diff/defaultMap.cy-default-map-story.png
