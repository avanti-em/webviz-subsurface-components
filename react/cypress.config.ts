import { defineConfig } from "cypress";
const getCompareSnapshotsPlugin = require("cypress-image-diff-js/dist/plugin");

export default defineConfig({
  component: {
    devServer: {
      framework: "create-react-app",
      bundler: "webpack",
    },
    video:false,
    setupNodeEvents(on, config) {
      getCompareSnapshotsPlugin(on, config);
    }
  },
});
