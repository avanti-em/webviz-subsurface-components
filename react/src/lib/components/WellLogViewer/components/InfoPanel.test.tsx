import { render, screen } from "@testing-library/react";
import "jest-styled-components";
import "@testing-library/jest-dom";
import React from "react";
import userEvent from "@testing-library/user-event";
import InfoPanel from "./InfoPanel";

describe("Test Info panel", () => {
    it("snapshot test", () => {
        const { container } = render(
            <InfoPanel
                header="Readout"
                infos={[
                    {
                        color: "black",
                        value: 2366,
                        type: "line",
                        track_id: "",
                        name: "DVER",
                        units: "M",
                    },
                    {
                        color: "",
                        value: Number.NaN,
                        type: "separator",
                        track_id: "",
                    },
                    {
                        color: "orange",
                        value: 8.5,
                        type: "line",
                        track_id: 11,
                    },
                    {
                        color: "green",
                        value: 17.57,
                        type: "line",
                        track_id: 9,
                    },
                ]}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
    it("snapshot test when value is infinity", () => {
        const { container } = render(
            <InfoPanel
                header="Readout"
                infos={[
                    {
                        color: "black",
                        value: Infinity,
                        type: "line",
                        track_id: "",
                        name: "DVER",
                        units: "M",
                    },
                    {
                        color: "",
                        value: Number.NaN,
                        type: "separator",
                        track_id: "",
                    },
                    {
                        color: "orange",
                        value: 8.5,
                        type: "line",
                        track_id: 11,
                    },
                    {
                        color: "green",
                        value: 17.57,
                        type: "line",
                        track_id: 9,
                    },
                ]}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
