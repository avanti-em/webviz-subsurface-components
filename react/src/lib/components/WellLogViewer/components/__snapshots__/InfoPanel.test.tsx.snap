// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Info panel snapshot test 1`] = `
<div>
  <fieldset>
    <legend>
      Readout
    </legend>
    <small>
      <table>
        <tbody>
          <tr>
            <td
              style="color: black;"
            >
              ⬤
            </td>
            <td>
              DVER
            </td>
            <td
              style="width: 80px; padding-left: 2em; text-align: right;"
            >
              2366
            </td>
            <td
              style="padding-left: 1em; font-size: x-small;"
            >
              M
            </td>
          </tr>
          <tr>
            <td
              colspan="4"
            >
               
              <hr />
            </td>
          </tr>
          <tr>
            <td
              style="color: orange;"
            >
              ⬤
            </td>
            <td />
            <td
              style="width: 80px; padding-left: 2em; text-align: right;"
            >
              8.5
            </td>
            <td
              style="padding-left: 1em; font-size: x-small;"
            />
          </tr>
          <tr>
            <td
              style="color: green;"
            >
              ⬤
            </td>
            <td />
            <td
              style="width: 80px; padding-left: 2em; text-align: right;"
            >
              17.57
            </td>
            <td
              style="padding-left: 1em; font-size: x-small;"
            />
          </tr>
        </tbody>
      </table>
    </small>
  </fieldset>
</div>
`;

exports[`Test Info panel snapshot test when value is infinity 1`] = `
<div>
  <fieldset>
    <legend>
      Readout
    </legend>
    <small>
      <table>
        <tbody>
          <tr>
            <td
              style="color: black;"
            >
              ⬤
            </td>
            <td>
              DVER
            </td>
            <td
              style="width: 80px; padding-left: 2em; text-align: right;"
            />
            <td
              style="padding-left: 1em; font-size: x-small;"
            >
              M
            </td>
          </tr>
          <tr>
            <td
              colspan="4"
            >
               
              <hr />
            </td>
          </tr>
          <tr>
            <td
              style="color: orange;"
            >
              ⬤
            </td>
            <td />
            <td
              style="width: 80px; padding-left: 2em; text-align: right;"
            >
              8.5
            </td>
            <td
              style="padding-left: 1em; font-size: x-small;"
            />
          </tr>
          <tr>
            <td
              style="color: green;"
            >
              ⬤
            </td>
            <td />
            <td
              style="width: 80px; padding-left: 2em; text-align: right;"
            >
              17.57
            </td>
            <td
              style="padding-left: 1em; font-size: x-small;"
            />
          </tr>
        </tbody>
      </table>
    </small>
  </fieldset>
</div>
`;
